tooling/minimize/src/bb.rs

```diff
@@ -70,19 +70,32 @@ fn translate_terminator<'cx, 'tcx>(
             ..
         } => translate_call(fcx, func, args, destination, target),
         rs::TerminatorKind::SwitchInt { discr, targets } => {
-            assert!(discr.ty(&fcx.body, fcx.cx.tcx).is_bool()); // for now we only support bool branching.
+            let ty = discr.ty(&fcx.body, fcx.cx.tcx);
 
-            let condition = translate_operand(discr, fcx);
-            let then_block = targets.target_for_value(1);
-            let then_block = fcx.bb_name_map[&then_block];
+            // For now we only support bool and int branching.
+            // FIXME: add support for switching on `char`
+            assert!(ty.is_bool() || ty.is_integral());
 
-            let else_block = targets.target_for_value(0);
-            let else_block = fcx.bb_name_map[&else_block];
+            // If the value is a boolean we need to cast it to an integer first as MiniRust switch only operates on ints.
+            let discr_op = translate_operand(discr, fcx);
+            let value = if ty.is_bool() {
+                ValueExpr::UnOp {
+                    operator: UnOp::BoolToIntCast(IntType { signed: Signedness::Unsigned, size: Size::from_bytes(1).unwrap() }),
```

Comment:
`<u8>::get_type()` would be easier than constructing the `IntType` by hand

Reply:
I agree, though I'd have to extract the `IntType` out of the returned `Type`. However this gives us a bit more future-proofness and readability so it is worth it in my opinion.

Reply:
Ah you are right, this needs an `IntType`, not a `Type`...

Can't this use the `if_` function from miniutil?

Reply:
> Can't this use the if_ function from miniutil?

Hm, that doesn't fit the MIR structure very well. Let's stick to what you got.